src/background-manager.ts

```diff
@@ -74,34 +76,35 @@ export class BackgroundManager {
   }
 
   public async addEntryToQueue<T>(entry: BackgroundEntry<T>, fireStartMessage?: boolean): Promise<string> {
-    let rval: string = null;
-    try {
-      const wrapped: InternalBackgroundEntry<T> = this.wrapEntryForInternal(entry);
-      rval = wrapped.guid;
-      if (this.localMode) {
-        Logger.info('Add entry to queue (local) : %j : Start : %s', entry, fireStartMessage);
-        this._localBus.next(wrapped);
-      } else {
-        // Guard against bad entries up front
-        const params = {
-          DelaySeconds: 0,
-          MessageBody: JSON.stringify(wrapped),
-          MessageGroupId: entry.type,
-          QueueUrl: this.awsConfig.queueUrl,
-        };
-
-        Logger.info('Add entry to queue (remote) : %j : Start : %s', params, fireStartMessage);
-        const result: AWS.SQS.SendMessageResult = await this.sqs.sendMessage(params).promise();
-
-        if (fireStartMessage) {
-          const fireResult: string = await this.fireStartProcessingRequest();
-          Logger.silly('FireResult : %s', fireResult);
-        }
+    const wrapped: InternalBackgroundEntry<T> = this.wrapEntryForInternal(entry);
+    const rval: string = wrapped.guid;
+    if (this.localMode) {
+      Logger.info('Add entry to queue (local) : %j : Start : %s', entry, fireStartMessage);
+      this._localBus.next(wrapped);
+    } else {
+      // Guard against bad entries up front
+      const params = {
+        DelaySeconds: 0,
+        MessageBody: JSON.stringify(wrapped),
+        MessageGroupId: entry.type,
+        QueueUrl: this.awsConfig.queueUrl,
+      };
+
+      Logger.info('Add entry to queue (remote) : %j : Start : %s', params, fireStartMessage);
+      const result: PromiseResult<AWS.SQS.Types.SendMessageResult, AWSError> = await this.sqs.sendMessage(params).promise();
+
+      const error = result?.$response?.error;
+      if (error) {
+        Logger.error('Error inserting background entry into SQS queue : %j', error);
+        throw new Error('Error inserting background entry into SQS queue : ' + error.code + ' : ' + error.name);
+      }
 
-        Logger.info('Background process %s using message id %s', rval, result.MessageId);
+      if (fireStartMessage) {
+        const fireResult: string = await this.fireStartProcessingRequest();
+        Logger.silly('FireResult : %s', fireResult);
       }
-    } catch (err) {
```

Comment:
I don't think we should be catching errors here. If it doesn't make it into the queue sending the GUI back anyways is confusing.